packages/eslint-plugin/src/rules/no-empty-styled-expression/__tests__/rule.test.ts

```diff
@@ -0,0 +1,75 @@
+import type { RuleTester } from 'eslint';
+
+import { typeScriptTester as tester } from '../../../test-utils';
+import { noEmptyStyledExpressionRule } from '../index';
+
+type InvalidTestCase = Omit<RuleTester.InvalidTestCase, 'errors'>;
+
+const createInvalidTestCases = (tests: InvalidTestCase[]) =>
+  tests.map((t) => ({
+    ...t,
+    errors: [{ messageId: 'unexpected' }],
+  }));
+
+tester.run('no-styled-tagged-template-expression', noEmptyStyledExpressionRule, {
+  valid: [
+    `
+        import { styled } from 'styled';
+
+        styled.div({
+            color: blue
+        });
+    `,
+    `
+        import { styled } from 'styled';
+
+        styled.span({
+            color: blue
+        });
+    `,
+    `
+        import { styled } from '@compiled/react';
+
+        styled.span("hello world");
+    `,
+    `
+        import { styled } from '@compiled/react';
+
+        styled.span({}, {});
```

Comment:
Thank you for catching this edge case! <3 Shouldn't we lint against these as well? I could be wrong here (not on laptop rn) but this seems equivalent to `styled.span({})`

Reply:
I did think so as well. @JakeLane mentioned different types of objects other than ObjectExpressions can be passed into the styled.span({}) call, but I don't know exactly what => thus he asked to only check for the case where it is (1) empty or (2) contains exactly one object, but the object is empty. Realistically, I don't know what edge cases people will attempt - this kind of blurs the line between using the function _**wrong**_ and using it a _**not recommended/best practice**_ way.

Reply:
yeah `styled` API is kinda scuffed in how it's a bit too flexible 🫠🫠🫠

happy to leave it as-is, since you mentioned on slack about how it would be too complex to handle all of the potential edge cases `styled` supports